### PR TITLE
Localization of 2 resource creation dialogs

### PR DESCRIPTION
**Pull request by taimir**
This is a follow up to https://github.com/kubernetes/dashboard/issues/742. I've extracted the text from the `Image Pull Secret` create dialog and `Namespace` create dialog on the deploy page for starters.

CC @bryk @floreks @maciaszczykm @digitalfishpond @zreigz @batikanu @f-higashi @cheld 

I'm putting everyone on CC. That way everyone will know the end result of this PR and how they should organize the text in their future commits. Also, here's a last chance to express any concerns with this approach :)

I plan on extracting all of the existing text in Dashboard in small pull requests like this one (step by step) so that we can review this more easily.

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/dashboard/766)

### CODE REVIEW

**Comment by taimir** [comment]
There is still one open end to this: how do we store text that is thoroughly reused throughout dashboard. E.g. `Create` or `Cancel` button labels. Those also need to be `goog.getMsg()`-ed, but currently I tend towards providing them globally to all dashboard pages. Opinions on this are welcome.

**Comment by cheld** [comment]
We should all use the same suffixes like _LABEL, _TITLE. These suffixes indicate the type of message. It is important for the translator to choose the right capitalization, gerund, .... 

Please make a very short md-file in the development section that explains the suffixes.

File: src/app/frontend/deploy/createsecret.html [hunk 1]
```diff
@@ -16,28 +16,28 @@
 
 <md-dialog aria-label="Create a new secret" layout="column" class="kd-create-secret-dialog">
   <md-dialog-content layout-padding>
-    <h4 class="md-title">Create a new image pull secret</h4>
-    <div>The new secret will be added to the cluster.</div>
+    <h4 class="md-title">{{ctrl.messages.MSG_IMAGE_PULL_SECRET_CREATE_DIALOG_TITLE}}</h4>
```
Comment (cheld) [inline src/app/frontend/deploy/createsecret.html]: what about shortening the messages variable:

ctrl.m.MSG_IMAGE_PULL_SECRET_CREATE_DIALOG_TITLE


ctrl.i18n.MSG_IMAGE_PULL_SECRET_CREATE_DIALOG_TITLE

**Comment by bryk** [comment]
> There is still one open end to this: how do we store text that is thoroughly reused throughout dashboard. E.g. Create or Cancel button labels. Those also need to be goog.getMsg()-ed, but currently I tend towards providing them globally to all dashboard pages. Opinions on this are welcome.

In other projects I worked on we had the messages copied, nothing was global. The reason is that in some languages "Cancel" on a form may be different than "Cancel" on a modal window.

File: src/app/frontend/deploy/createsecret.html (hunk 1 above)
Comment (bryk) [inline src/app/frontend/deploy/createsecret.html]: `i18n` looks nice.

**Comment by bryk** [comment]
Review status: 0 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/app/frontend/deploy/createsecret.html, line 25 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIWau17xZcStskvdNRh:-KIWau180VGrU5Flmvhx:2067681283) ([raw file](https://github.com/kubernetes/dashboard/blob/4799cc3d523bddec9353e433c481e7ce0f4ece0c/src/app/frontend/deploy/createsecret.html#L25)):*
> ```HTML
>       <kd-help-section>
>         <md-input-container class="md-block" layout-wrap>
>           <label>{{ctrl.messages.MSG_IMAGE_PULL_SECRET_NAME_LABEL}}</label>
> ```

How about using one-time binding for all messages? I.e.,  `{{::ctrl.messages.MSG_IMAGE_PULL_SECRET_NAME_LABEL}}`. This is faster than normal bindings and the messages do not change.


How about we enforce it via documentation and code style? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766)*

**Comment by bryk** [comment]
Review status: 0 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

*[src/app/frontend/deploy/createsecret_controller.js, line 127 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIWbJuoVPSSAiv96Kv7:-KIWbJuoVPSSAiv96Kv8:-461642936) ([raw file](https://github.com/kubernetes/dashboard/blob/4799cc3d523bddec9353e433c481e7ce0f4ece0c/src/app/frontend/deploy/createsecret_controller.js#L127)):*
> ```JavaScript
> }
> 
> const createSecretMessages = {
> ```

IMO, you can name it just "messages", or  "i18n". 

---

*[src/app/frontend/deploy/createsecret_controller.js, line 128 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIWbOaKMQodmTKveSKM:-KIWbOaKMQodmTKveSKN:1700543660) ([raw file](https://github.com/kubernetes/dashboard/blob/4799cc3d523bddec9353e433c481e7ce0f4ece0c/src/app/frontend/deploy/createsecret_controller.js#L128)):*
> ```JavaScript
> 
> const createSecretMessages = {
>   /** @export @type string @desc Create image pull secret dialog title. The message appears at the top of the dialog box. */
> ```

You dont need `@type` here. Just `@export {string}`. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766)*

**Comment by taimir** [comment]
OK, thanks for the comments. I'll change it later this evening :)

**Comment by taimir** [comment]
PTAL, I've updated the `MSG_` variables and added a documentation draft.

<details><summary><i>Previously, codecov-io wrote…</i></summary>
> ## [Current coverage][cc-pull] is **95.20%**
> > Merging [#766][cc-pull] into [master][cc-base-branch] will increase coverage by **0.43%**
> 
> 1. File `...ereader_directive.js` (not in diff) was modified. [more](https://codecov.io/gh/kubernetes/dashboard/commit/16856a02ea13598ad439a2f81c0954f0d1d44a5b/changes?src=pr#7372632F6170702F66726F6E74656E642F6465706C6F792F66696C657265616465725F6469726563746976652E6A73) 
>   - Misses `-6` 
>   - Partials `0` 
>   - Hits `+6`
> 
> ```diff
> @@             master       #766   diff @@
> ==========================================
>   Files           177        177          
>   Lines          1392       1396     +4   
>   Methods           0          0          
>   Messages          0          0          
>   Branches          0          0          
> ==========================================
> + Hits           1319       1329    +10   
> + Misses           73         67     -6   
>   Partials          0          0          
> ```
> 
> > Powered by [Codecov](https://codecov.io?src=pr). Last updated by [7df7b15...16856a0][cc-compare]
> [cc-base-branch]: https://codecov.io/gh/kubernetes/dashboard/branch/master?src=pr
> [cc-compare]: https://codecov.io/gh/kubernetes/dashboard/compare/7df7b152137645364281758834795f41c5170640...16856a02ea13598ad439a2f81c0954f0d1d44a5b
> [cc-pull]: https://codecov.io/gh/kubernetes/dashboard/pull/766?src=pr
</details>


---

Review status: 0 of 9 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-:-KIbLUrgpz-gBjj5qnwO:2075329955)*

**Comment by cheld** [comment]
Very nice.

LGTM

**Comment by bryk** [comment]
Last cosmetic comments and we can merge. OK? :) 

<details><summary><i>Previously, cheld (Christoph Held) wrote…</i></summary>
> Very nice.
> 
> 
> 
> LGTM
</details>


---

Reviewed 1 of 7 files at r1, 8 of 8 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[docs/devel/localization.md, line 16 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIbO7_L1rs0F5abcZxo:-KIbO7_L1rs0F5abcZxp:-895643821) ([raw file](https://github.com/kubernetes/dashboard/blob/2367fbd886ac422ddd319eb0fbb50dc532a374b8/docs/devel/localization.md#L16)):*
> ```Markdown
> 
> ```{js}
> /** @export string @desc The text appears when the image pull secret name exceeds the maximal length. */
> ```

Missing braces around `string`. 

---

*[docs/devel/localization.md, line 40 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIbO3cduYTJsF0EHoxz:-KIbO3cduYTJsF0EHoy-:-532686854) ([raw file](https://github.com/kubernetes/dashboard/blob/2367fbd886ac422ddd319eb0fbb50dc532a374b8/docs/devel/localization.md#L40)):*
> ```Markdown
>     ...
> 
>     this.i18n = i18n;
> ```

Can you add `@export` here? 

---

*[src/app/frontend/deploy/createnamespace_controller.js, line 72 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIbNy9CYRrnEX1updHx:-KIbNy9CYRrnEX1updHy:-985938047) ([raw file](https://github.com/kubernetes/dashboard/blob/2367fbd886ac422ddd319eb0fbb50dc532a374b8/src/app/frontend/deploy/createnamespace_controller.js#L72)):*
> ```JavaScript
>     /**
>      * @export
>      * {!Object<string, string>}
> ```

How about you omit type here? Just `@export`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-:-KIbOBbmV-xg06Xkr-3n:-1684519005)*

**Comment by taimir** [comment]
Review status: 4 of 9 files reviewed at latest revision, 4 unresolved discussions.

---

*[docs/devel/localization.md, line 16 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIbO7_L1rs0F5abcZxo:-KIbeyVMVOi1LfGJ8waz:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/2367fbd886ac422ddd319eb0fbb50dc532a374b8/docs/devel/localization.md#L16)):*
<details><summary><i>Previously, bryk (Piotr Bryk) wrote…</i></summary>
> Missing braces around `string`. 
</details>
Done.

---

*[docs/devel/localization.md, line 40 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIbO3cduYTJsF0EHoxz:-KIbgfzN619Hth5m3Iny:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/2367fbd886ac422ddd319eb0fbb50dc532a374b8/docs/devel/localization.md#L40)):*
<details><summary><i>Previously, bryk (Piotr Bryk) wrote…</i></summary>
> Can you add `@export` here? 
</details>
Done.

---

*[src/app/frontend/deploy/createnamespace_controller.js, line 72 \[r2\]](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-KIbNy9CYRrnEX1updHx:-KIbgh1EsDZF9nClgqaX:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/2367fbd886ac422ddd319eb0fbb50dc532a374b8/src/app/frontend/deploy/createnamespace_controller.js#L72)):*
<details><summary><i>Previously, bryk (Piotr Bryk) wrote…</i></summary>
> How about you omit type here? Just `@export`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766)*

**Comment by bryk** [comment]
Reviewed 5 of 5 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766)*

**Comment by bryk** [comment]
Travis has problems. Is this flake or real test failures? 

<details><summary><i>Previously, bryk (Piotr Bryk) wrote…</i></summary>
> Last cosmetic comments and we can merge. OK? :) 
</details>


---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-:-KIbrNWO-PQkH4fhX0SN:875562418)*

**Comment by taimir** [comment]
It seemed like a flake to me, if I didn't miss some obscure detail. 

<details><summary><i>Previously, bryk (Piotr Bryk) wrote…</i></summary>
> Travis has problems. Is this flake or real test failures? 
</details>


---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-:-KIbs5AjSgbBL1BiG776:960301925)*

**Comment by bryk** [comment]
can you run gulp integration-test? It fails. 

<details><summary><i>Previously, taimir (Atanas Mirchev) wrote…</i></summary>
> It seemed like a flake to me, if I didn't miss some obscure detail. 
</details>


---







*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-:-KIbvD8cYrCoRDSRxnn8:1143086145)*

**Comment by taimir** [comment]
Strange ... On my machine it runs to completion:


<details><summary><i>Previously, bryk (Piotr Bryk) wrote…</i></summary>
> can you run gulp integration-test? It fails. 
</details>


---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-:-KIbwAyYDJ7CqQU5ygdI:1569774411)*

**Comment by taimir** [comment]
OK, now the tests are passing - as you can see, I just changed two words in the dev doc... hah

PTAL

<details><summary><i>Previously, taimir (Atanas Mirchev) wrote…</i></summary>
> Strange ... On my machine it runs to completion:
> 
> 
</details>


---

Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-:-KIcECZBNLJy2BKTeYow:1166872184)*

**Comment by bryk** [comment]
:lgtm: Perfect! 

<details><summary><i>Previously, taimir (Atanas Mirchev) wrote…</i></summary>
> OK, now the tests are passing - as you can see, I just changed two words in the dev doc... hah
> 
> PTAL
</details>


---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/766#-:-KJ-Ta_CJ4FiWu7LoZQu:-1729251779)*

**Pull request merged**